### PR TITLE
Update user data

### PR DESCRIPTION
**Pull request by Rastopapola**
Hello simple-sso team,

it's me again! Here is an implementation, coming with a test for issue #61 . 

It would love to hear some feedback soon. 

Best regards!

### CODE REVIEW

**Comment by Rastopapola** [comment]
Sorry, I forgot to mention some people: I summon @GaretJax or @kinkerl 

Thanks!

File: simple_sso/sso_client/client.py [hunk 1]
```diff
@@ -108,6 +108,10 @@ def get_user(self, access_token):
     def build_user(self, user_data):
         try:
             user = User.objects.get(username=user_data['username'])
+            # Update user data, excluding username changes
+            del user_data['username']
```
Comment (GaretJax) [inline simple_sso/sso_client/client.py:112]: This modifies the `user_data` dictionary that has been created by the caller and is not ideal.
A better approach is to check the keys in the for loop.

I'm also not sure about the security implications. What does user_data contain? Are all attributes meant to be reset?

**Review by GaretJax — CHANGES_REQUESTED** [review]

**Comment by GaretJax** [comment]
Thanks for your PR @Rastopapola :-)

I left a comment.

**Comment by Rastopapola** [comment]
@GaretJax 
Sure thing! I really enjoy your project, which helps me a lot as well and I would be proud to contribute something practical :)

I see your concerns regarding the user_data. To decrease the amount of if-checks inside the loop I suggest building a 'user_data_tmp' dict beforehand, excluding the username. This way 'user_data' won't be touched and we don't have redundant checks in the loop.

'user_data' is the serialized 'User' model which is received by the sso client application for e.g. creating a new user. It holds only values that are provided by the sso server: https://github.com/divio/django-simple-sso/blob/master/simple_sso/sso_server/server.py#L153-L166 
As you can see there is some 'extra_data' which can extend the serialized basic 'User' model data. As long as there are no critical extra-user-data, which has been added there by the developer inside the project, this critical data won't be received by any client.

I quickly update the code on this PR, so you can take another look ;)

File: simple_sso/sso_client/client.py (hunk 1 above)
Comment (Rastopapola) [reply simple_sso/sso_client/client.py]: Good idea, keeping the original 'user_data'. I just pushed your requested change: https://github.com/divio/django-simple-sso/pull/62/commits/a0f864d973b6db9b238a86284847f185e8e854a7#diff-156d4d0f83e2028594d28cfe5b57a3aaec983a3722c45c6f9aaf0c7619a37c1dL112-R116 

Answering your question in here as well:
1. 'user_data' contains the serialized data from the sso server's 'User' object, extended with some extra_data, as seen in https://github.com/divio/django-simple-sso/blob/master/simple_sso/sso_server/server.py#L153-L166
2. All attributes beside the username are meant to be updated according to the user_data, which is provided on every user's login. This way the sso client will always have the updated user data from the sso server the next time the user logs into the sso client.
3. Regarding security implications: Of course one could add a set of 'editable_user_attrs' which defines the attributes, which can be updated on a login. But that is not needed when the project developers make sure no critical data is added to the 'extra_data' dict on the sso server's side, imho.

**Comment by Rastopapola** [comment]
@GaretJax 
Your change request has been pushed :)

**Comment by Rastopapola** [comment]
@FinalAngel @GaretJax 
Ping! ;)
Any news or thoughts on this?

**Comment by GaretJax** [comment]
Looks good (note that `copy` iterates over the dict, so we now have two loops, instead of a single loop with an if, but it should not matter performance-wise).

**Review by GaretJax — APPROVED** [review]

**Comment by Rastopapola** [comment]
> Looks good (note that `copy` iterates over the dict, so we now have two loops, instead of a single loop with an if, but it should not matter performance-wise).

Oh, my humble apology on this. I was hoping for some high performance pointer C magic in the background. Thanks for the hint, I will remember this the next time. 

And thanks for approving the pull request! Feeling pretty honored now, since that's my first approved open source contribution on another project. 
When can we expect the changes being available via `pip`? :)

**Comment by Rastopapola** [comment]
@FinalAngel  @GaretJax 
Sorry for another disturbance, but I wanted to ask when the PR will be merged, so the feature will be available via pip?

**Comment by GaretJax** [comment]
This has been released as 1.1.0

**Comment by Rastopapola** [comment]
Thank you so much!

**Pull request merged**